source/frontend/src/features/mapSideBar/acquisition/tabs/compensation/detail/CompensationRequisitionDetailView.tsx

```diff
@@ -120,18 +119,28 @@ export const CompensationRequisitionDetailView: React.FunctionComponent<
 
   const payeeDetails = getPayeeDetails(compensationPayee);
 
+  const editButtonBlock = (
+    <EditButton
+      title="Edit compensation requisition"
+      onClick={() => {
+        setEditMode(true);
+      }}
+    />
+  );
   return (
     <StyledSummarySection>
       <LoadingBackdrop show={loading} parentScreen={true} />
       <RightFlexDiv>
-        {setEditMode !== undefined && hasClaim(Claims.COMPENSATION_REQUISITION_EDIT) && (
-          <EditButton
-            title="Edit compensation requisition"
-            onClick={() => {
-              setEditMode(true);
-            }}
-          />
-        )}
+        {setEditMode !== undefined &&
```

Comment:
could we have this refactored so that there is a single block that conditionally renders the edit button? we should never have a case where multiple edit buttons are returned, so I'd rather the render logic more clearly reflected that.

Reply:
maybe turn these if statements into named functions, so we can do something like: 
if(isDraftAndCanEdit || isNotDraftAndAdmin) or similar

Reply:
updated to use bool variables.